### PR TITLE
changes to allow collecting builds from a workflow type jenkins job

### PR DESCRIPTION
**Pull request by Mpluya**
Problem: When the Jenkins collector runs and encounters a workflow type job ("_class":"org.jenkinsci.plugins.workflow.job.WorkflowRun"), the collector throws an exception since workflow type jobs don't have changeSets. This is a fix to add a null check and immediately exit from the method when there are no changeSets to build, but still allow to collect build information from a workflow job.

### CODE REVIEW

**Comment by amitmawkin** [comment]
@Mpluya Does this take care for Jenkins 2.0 pipelines too, or just the workflow plugin.

File: collectors/build/jenkins/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java [hunk 1]
```diff
@@ -232,6 +233,7 @@ public static String rebuildJobUrl(String build, String server) throws URISyntax
      */
     private void addChangeSets(Build build, JSONObject buildJson) {
         JSONObject changeSet = (JSONObject) buildJson.get("changeSet");
+        if(changeSet == null) { return; }
```
Comment (MarkRx) [inline collectors/build/jenkins/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: This only prevents a NPE. It does not correctly gather information

**Comment by MarkRx** [comment]
@pxk5958 has a PR coming soon that will support the pipeline plugin

**Comment by amitmawkin** [comment]
Thanks @MarkRx

**Comment by MarkRx** [comment]
#1038

**Comment by Mpluya** [comment]
@amitmawkin this solves the error encountered on any workflow type job with just an addition of a null check. I probably should reword this request to instead say "handle null changeSet scenarios",
 which i happen to have encountered on workflow type jobs. Thanks.

**Comment by pxk5958** [comment]
@Mpluya Workflow/Pipeline jobs have the changesets as a JSONArray of "changeSets" instead of "changeSet" in normal jobs. That's why you're encountering the NPE. With your change, it would skip the NPE, but will not collect the actual changesets if present.

**Comment by Mpluya** [comment]
@pxk5958 I appreciate the explanation. Closing this request since PR #1038 addresses the entire need.

**Pull request closed without merging**